### ISSUE TITLE
[Question]: elastic_transport.ConnectionError: Connection error caused by: NameResolutionError

### ISSUE THREAD

**Issue by Said-Apollo** [infiniflow/ragflow#1523]
### Describe your problem

Hi, Im still encountering the following error, when trying to run bash ./entrypoint.sh
(ragflow) said@said-HP-Laptop-17-cp3xxx:~/ragflow$ bash ./entrypoint.sh 
`  File "/home/said/anaconda3/envs/ragflow/lib/python3.11/site-packages/elastic_transport/_node/_http_urllib3.py", line 202, in perform_request
    raise err from None
elastic_transport.ConnectionError: Connection error caused by: NameResolutionError(<urllib3.connection.HTTPConnection object at 0x70d7591af510>: Failed to resolve 'es01' ([Errno -3] Temporary failure in name resolution))`

WHen doing docker ps, it looks like this
`CONTAINER ID   IMAGE                                                  COMMAND                  CREATED         STATUS                   PORTS                              NAMES
ddd2663ea0ac   mysql:5.7.18                                           "docker-entrypoint.s…"   4 minutes ago   Up 4 minutes (healthy)   0.0.0.0:5455->3306/tcp             ragflow-mysql
d0e1c727036a   quay.io/minio/minio:RELEASE.2023-12-20T01-00-02Z       "/usr/bin/docker-ent…"   4 minutes ago   Up 4 minutes             0.0.0.0:9000-9001->9000-9001/tcp   ragflow-minio
a410df681c58   redis:7.2.4                                            "docker-entrypoint.s…"   4 minutes ago   Up 4 minutes             0.0.0.0:6379->6379/tcp             ragflow-redis
335e6e9d7601   docker.elastic.co/elasticsearch/elasticsearch:8.11.3   "/bin/tini -- /usr/l…"   4 minutes ago   Up 4 minutes (healthy)   9300/tcp, 0.0.0.0:1200->9200/tcp   ragflow-es-01`

Afterwards I also tried to change the Port from 1200 to 9200 in the .env file. I deleted docker container and restarted again. Doing again docker ps gave me
`(ragflow) said@said-HP-Laptop-17-cp3xxx:~/ragflow/docker$ docker ps
CONTAINER ID   IMAGE                                                  COMMAND                  CREATED              STATUS                        PORTS                              NAMES
5f021df0e272   mysql:5.7.18                                           "docker-entrypoint.s…"   About a minute ago   Up About a minute (healthy)   0.0.0.0:5455->3306/tcp             ragflow-mysql
5a8e72f38440   redis:7.2.4                                            "docker-entrypoint.s…"   About a minute ago   Up About a minute             0.0.0.0:6379->6379/tcp             ragflow-redis
6d2a7255435f   docker.elastic.co/elasticsearch/elasticsearch:8.11.3   "/bin/tini -- /usr/l…"   About a minute ago   Up About a minute (healthy)   0.0.0.0:9200->9200/tcp, 9300/tcp   ragflow-es-01
963f11809f66   quay.io/minio/minio:RELEASE.2023-12-20T01-00-02Z       "/usr/bin/docker-ent…"   About a minute ago   Up About a minute             0.0.0.0:9000-9001->9000-9001/tcp   ragflow-minio`

Again, trying to run bash ./entrypoint.sh again gave me the same error
`(ragflow) said@said-HP-Laptop-17-cp3xxx:~/ragflow$ bash ./entrypoint.sh 
nginx: [alert] could not open error log file: open() "/var/log/nginx/error.log" failed (13: Permission denied)
2024/07/15 19:06:56 [warn] 1702390#1702390: the "user" directive makes sense only if the master process runs with super-user privileges, ignored in /etc/nginx/nginx.conf:1
2024/07/15 19:06:56 [emerg] 1702390#1702390: open() "/var/log/nginx/access.log" failed (13: Permission denied)
[WARNING] [2024-07-15 19:06:56,956] [_node_pool.mark_dead] [line:249]: Node <Urllib3HttpNode(http://es01:9200)> has failed for 1 times in a row, putting on 1 second timeout
[WARNING] [2024-07-15 19:06:56,956] [_transport.perform_request] [line:395]: Retrying request after failure (attempt 0 of 3)
Traceback (most recent call last):
  File "/home/said/anaconda3/envs/ragflow/lib/python3.11/site-packages/elastic_transport/_transport.py", line 328, in perform_request
    meta, raw_data = node.perform_request(
                     ^^^^^^^^^^^^^^^^^^^^^
  File "/home/said/anaconda3/envs/ragflow/lib/python3.11/site-packages/elastic_transport/_node/_http_urllib3.py", line 202, in perform_request
    raise err from None
elastic_transport.ConnectionError: Connection error caused by: NameResolutionError(<urllib3.connection.HTTPConnection object at 0x721cc3f9dfd0>: Failed to resolve 'es01' ([Errno -3] Temporary failure in name resolution))
[WARNING] [2024-07-15 19:06:56,957] [_node_pool.mark_dead] [line:249]: Node <Urllib3HttpNode(http://es01:9200)> has failed for 2 times in a row, putting on 2 second timeout
[WARNING] [2024-07-15 19:06:56,957] [_transport.perform_request] [line:395]: Retrying request after failure (attempt 1 of 3)
Traceback (most recent call last):
  File "/home/said/anaconda3/envs/ragflow/lib/python3.11/site-packages/elastic_transport/_transport.py", line 328, in perform_request
    meta, raw_data = node.perform_request(
                     ^^^^^^^^^^^^^^^^^^^^^
  File "/home/said/anaconda3/envs/ragflow/lib/python3.11/site-packages/elastic_transport/_node/_http_urllib3.py", line 202, in perform_request
    raise err from None
elastic_transport.ConnectionError: Connection error caused by: NameResolutionError(<urllib3.connection.HTTPConnection object at 0x721cc3f8df10>: Failed to resolve 'es01' ([Errno -3] Temporary failure in name resolution))
[WARNING] [2024-07-15 19:06:56,958] [_node_pool.mark_dead] [line:249]: Node <Urllib3HttpNode(http://es01:9200)> has failed for 3 times in a row, putting on 4 second timeout
[WARNING] [2024-07-15 19:06:56,958] [_transport.perform_request] [line:395]: Retrying request after failure (attempt 2 of 3)
Traceback (most recent call last):
  File "/home/said/anaconda3/envs/ragflow/lib/python3.11/site-packages/elastic_transport/_transport.py", line 328, in perform_request
    meta, raw_data = node.perform_request(
                     ^^^^^^^^^^^^^^^^^^^^^
  File "/home/said/anaconda3/envs/ragflow/lib/python3.11/site-packages/elastic_transport/_node/_http_urllib3.py", line 202, in perform_request
    raise err from None
elastic_transport.ConnectionError: Connection error caused by: NameResolutionError(<urllib3.connection.HTTPConnection object at 0x721cc3f90510>: Failed to resolve 'es01' ([Errno -3] Temporary failure in name resolution))
`

**Comment by Said-Apollo** [comment]
Going to localhost:9200 and logging in results in this message

![image](https://github.com/user-attachments/assets/d00005b6-1a60-47aa-af57-d20d514207b2)

**Comment by KevinHuSh** [comment]
Check the host parts in conf/service_conf.ymal if you're not using docker to deply.

**Comment by Said-Apollo** [comment]
Nice, it works. However, for some reason I had to change not only the Hosts in conf/service_conf.yml but also the ones in docker/service_conf.yml
Also, in the .env file the es port was initially set to 1200, which I set to 9200  (otherwise it did not work for me, not sure why). Lastly, when adding the hosts, e.g. all had 127.0.0.1, however, when doing the same for mysql it failed, but it worked when using localhost for mysql as host.
